### PR TITLE
Disposal units call dropped() on items placed inside them

### LINKED ISSUE #37629: Mulligan syringe's "Draw" text is ivisble as a real-word, non-abstract spirte
The round ID is: 87335
image

### PR DESCRIPTION
**Pull request by harmonyn**
Fixes https://github.com/tgstation/tgstation/issues/37629. 

Basically: The DRAW/INJECT is an overlay; syringes have update_icon() called in dropped(), but disposal bins use a snowflake proc that previously never called it, hence the overlay not going away 

Possibly fixes other stuff as well I haven't read all 1,120 issues

### CODE REVIEW

**Comment by vuonojenmustaturska** [comment]
`dropped()` _must_ happen for items that leave the inventory

**Comment by bgobandit** [comment]
it does normally (hence why this bug doesn't appear with closets or crates) but disposal bins have a snowflake place_item_in_disposal proc that bypasses it

**Comment by SpaceManiac** [comment]
What Naksu is trying to say is that arranging for `dropped()` to be called would be a better fix.

**Review by Cyberboss — COMMENTED** [review]
dropped() should be called already when temporarilyRemoveItemFromInventory is called on line 99. I highly doubt this is correct

**Comment by bgobandit** [comment]
I mean, it isn't. It can't be because update_icon() is called upon the dropped proc for syringes, yet it never happens:

`/obj/item/reagent_containers/syringe/dropped(mob/user)
	..()
	update_icon()`

**Comment by Cyberboss** [comment]
untitled

File: code/modules/recycling/disposal/bin.dm [hunk 1]
```diff
@@ -106,6 +106,7 @@
 
 /obj/machinery/disposal/proc/place_item_in_disposal(obj/item/I, mob/user)
 	I.forceMove(src)
+	I.dropped(user)
 	user.visible_message("[user.name] places \the [I] into \the [src].", "<span class='notice'>You place \the [I] into \the [src].</span>")
```
Comment (SpaceManiac) [inline code/modules/recycling/disposal/bin.dm]: Can you do me a favor while you're here and change `[user.name]` to just `[user]` on this line?

**Comment by bgobandit** [comment]
@Cyberboss is there supposed to be a runtime on that

also can you do it with a syringe not a piece of paper

**Comment by AnturK** [comment]
Means the cause here is in dropped is called with mob as location since TemporarilyRemove does not change the item's location. You need to update syringe inventory detection. We have a flag that's toggled on items in inventory you can use that.

**Comment by bgobandit** [comment]
I can't find the flag

**Comment by AnturK** [comment]
`IN_INVENTORY`

**Pull request closed without merging**